cmd/helm/dependency.go

```diff
@@ -71,6 +71,36 @@ the dependency charts stored locally. The path should start with a prefix of
 If the dependency chart is retrieved locally, it is not required to have the
 repository added to helm by "helm add repo". Version matching is also supported
 for this case.
+
+A repository can be defined as a git URL. The path must start with a prefix of
+"git:" followed by a valid git repository URL.
+
+    # Chart.yaml
+    dependencies:
+    - name: helm-chart
+      version: "main"
+      repository: "git:https://github.com/helm/helm-chart.git"
```

Comment:
If https is only supported, this should just be:
```
git://github.com/helm/helm-chart
```

Is ssh supported? What happens if my private key is password protected?

Reply:
ssh is supported now.

Reply:
now, Similar to the performance of the git command line.